### DISCUSSION TITLE
Acesso a dados com Dapper: Dúvida de SQL comando com JOIN

### DISCUSSION THREAD

**Discussion by oimarcella** [balta-io/forum#560] [Tira Dúvidas] [answered]
Na aula sobre repositórios especialistas Balta faz um SQL que eu entendi bem.
Era para selecionar tudo de role e de user qu estivesse relacionado, e mesmo os usuários que não tinham roles.
```
SELECT [User].*, [Role].*
                FROM [User]
                LEFT JOIN [UserRole] ON [UserRole].[UserId] = [User].[Id]
                LEFT JOIN [Role] ON [Role].[Id] = [UserRole].[RoleId]
```
Eu queria fazer o mesmo so que invertendo, puxar as roles e usuários relacionados, e também os roles que não estavam em uso. Pensei da seguine forma:

```
SELECT [Role].*, [User].*
                FROM [Role]
                LEFT JOIN [UserRole] ON [UserRole].[UserId] = [User].[Id]
                LEFT JOIN [User] ON [User].[Id] = [UserRole].[UserId]
```

Porém não funcionou, ele continua trazendo o mesmo resultado a primeira query

**Comment by andrebaltieri** [comment]
@oimarcella como vai?

Na verdade funcionou, é que os dados e a relação é a mesma... Tanto os roles quanto os usuários continuam com a mesma relação, independe de você obter os perfis ou usuários primeiro.

Talvez o que você queira seja popular um objeto `Role` com uma lista de `User`? Seria isto?
Como está seu código C# que faz a leitura desse SQL?

A propósito, parabéns pelo empenho e evolução nos estudos 🚀

**Reply by oimarcella** [reply]
Olá @andrebaltieri, obrigada pela resposta.

Não sei se entendi bem sua pergunta, mas vou tentar mostrar de forma mais visual oque eu quero dizer.

Então, eu não cheguei a fazer a leitura desse SQL no momento da pergunta, eu executei apenas pelo Azure Data Studio.

Então, pelo que você disse, eu acho que então eu não fiz oque eu queria fazer.

Primeira query me retornou (resumidamente):
Perceba que me retornou os usuários e suas roles, e mesmo os usuários sem roles
```
1 Rafael rafael@email.com | author
2 Marcella marcella@email.com
3 Balta balta@email.com | admin
4 Larissa larissa@email.com | author
```

Eu gostaria de entender como seria para ter  seguinte resultado:
As roles, e seus usuários, e também as roles que estão sem uso.
```
1  author | Larissa
1  author | Rafael
2 user
3 admin | Balta
4 student
```

**Reply by andrebaltieri** [reply, marked as answer]
Bom dia, @oimarcella, como vai?

Pelo seu resultado, só mudou a ordem, é isto mesmo? Sua select já está com o JOIN correto!

```sql

SELECT [Role].[Name], [User].[Name]
                FROM [Role]
                LEFT JOIN [UserRole] ON [UserRole].[UserId] = [User].[Id]
                LEFT JOIN [User] ON [User].[Id] = [UserRole].[UserId]
```

**Reply by oimarcella** [reply]
Agradeço
Vou adicionar mais dados para entender, talvez facilite enxergar a relação.
